pkg/config/v2/parameter/compound/compound.go

```diff
@@ -0,0 +1,165 @@
+	formatInterface, ok := context.Value["format"]
+	if !ok {
+		return nil, createParameterParseError(context, "missing property `format`")
+	}
+
+	format, err := template.New(context.ParameterName).Option("missingkey=error").Parse(util.ToString(formatInterface))
+	if err != nil {
+		return nil, createParameterParseError(context, "format is not a valid template: %v", err)
+	}
+
+	references, ok := context.Value["references"]
+	if !ok {
+		return nil, createParameterParseError(context, "missing property `references`")
+	}
+
+	referencedParameterSlice, ok1 := references.([]interface{})
+	if !ok1 {
+		return nil, createParameterParseError(context, "malformed value `references`")
+	}
+	referencedParameters := toParameterReferences(referencedParameterSlice, context.Coordinate)
+
+	return &CompoundParameter{
+		Format:               format,
+		ReferencedParameters: referencedParameters,
+	}, nil
+}
+
+func writeCompoundParameter(context parameter.ParameterWriterContext) (map[string]interface{}, error) {
+	compoundParam, ok := context.Parameter.(*CompoundParameter)
+
+	if !ok {
+		return nil, &parameter.ParameterWriterError{
+			Location: context.Coordinate,
+			EnvironmentDetails: errors.EnvironmentDetails{
+				Group:       context.Group,
+				Environment: context.Environment,
+			},
+			ParameterName: context.ParameterName,
+			Reason:        "unexpected type. parameter is not of type `CompoundParameter`",
+		}
+	}
+
+	result := make(map[string]interface{})
+
+	format := util.ToString(compoundParam.Format)
+	if format == "" {
+		return nil, createParameterWriteError(context, "missing property `format`")
+	}
+	result["format"] = format
+
+	if len(compoundParam.ReferencedParameters) == 0 {
+		return nil, createParameterWriteError(context, "missing property `references`")
+	}
+	result["format"] = compoundParam.ReferencedParameters
+
+	return result, nil
+}
+
+func createParameterParseError(context parameter.ParameterParserContext, reason string, a ...interface{}) error {
+	return &parameter.ParameterParserError{
+		Location: context.Coordinate,
+		EnvironmentDetails: errors.EnvironmentDetails{
+			Group:       context.Group,
+			Environment: context.Environment,
+		},
+		ParameterName: context.ParameterName,
+		Reason:        fmt.Sprintf(reason, a...),
+	}
+}
+
+func createParameterWriteError(context parameter.ParameterWriterContext, reason string, a ...interface{}) error {
+	return &parameter.ParameterWriterError{
+		Location: context.Coordinate,
+		EnvironmentDetails: errors.EnvironmentDetails{
+			Group:       context.Group,
+			Environment: context.Environment,
+		},
+		ParameterName: context.ParameterName,
+		Reason:        fmt.Sprintf(reason, a...),
+	}
+}
+
+func toParameterReferences(params []interface{}, coord coordinate.Coordinate) (paramRefs []parameter.ParameterReference) {
+	for _, param := range params {
+		paramRefs = append(paramRefs, parameter.ParameterReference{
+			Config:   coord,
+			Property: util.ToString(param),
```

Comment:
This is a bit odd if you ask me. It would be possible to 
use something like
```yaml
references:
 - name: test
   age: test
 - name: hello
```

which never makes sense. We might want to 
switch on the type of the param here and return an
error in case it is either a map or an array.

Reply:
That's true, but shouldn't we just go all the way then and just only allow strings in this place and fail in any other case?

Reply:
I would agree if the config file format was not YAML :laughing: 
Otherwise one would be required to always add quotes around
the list of references. 

Hence my suggestion of treating everything besides arrays and 
maps as strings.